extensions/slack/src/shared/client/SlackClient.ts

```diff
@@ -0,0 +1,162 @@
+import { slackWebClient } from "./WebClient";
+import { LocalStorageService } from "./LocalStorageService";
+
+interface Item {
+  id: string;
+  teamId: string;
+  name: string;
+}
+
+export interface User extends Item {
+  username: string;
+  icon: string | undefined;
+}
+
+export type Channel = Item;
+export type Group = Item;
+
+export type PresenceStatus = "online" | "offline" | "forced-offline";
+export interface SnoozeStatus {
+  nextDndEnd: Date | undefined;
+  snoozeEnd: Date | undefined;
+}
+
+const sortNames = (a: string, b: string) => {
+  if (a < b) {
+    return -1;
+  }
+  if (a > b) {
+    return 1;
+  }
+  return 0;
+};
+
+export class SlackClient {
+  public static async getUsers(): Promise<User[]> {
+    const users = await LocalStorageService.get<User[]>("users", async (): Promise<User[]> => {
```

Comment:
What if accessToken is updated in preferences and the new token happens to be from a different Slack workspace (users / channels are different)?
I think caching is okay but they should be prefixed (somehow) in a way that changing access token doesn't display cached data from a different Slack workspace

Reply:
Should be fixed with my new changes since I'm now using the package `swr` for caching.